### PR TITLE
fix(document_loaders/telegram): fix pandas calls + add tests

### PR DESCRIPTION
**Pull request by 0xRaduan**
# Fix Telegram API loader + add tests.
I was testing this integration and it was broken with next error:
```python
message_threads = loader._get_message_threads(df)
KeyError: False
```
Also, this particular loader didn't have any tests / related group in poetry, so I added those as well.

@hwchase17 / @eyurtsev please take a look on this fix PR.

### CODE REVIEW

**Comment by 0xRaduan** [comment]
Let me fix type hints for telegram as well.

File: langchain/document_loaders/telegram.py [hunk 1]
```diff
@@ -82,24 +82,26 @@ class TelegramChatApiLoader(BaseLoader):
 
     def __init__(
         self,
-        chat_url: Optional[str] = None,
+        chat_entity: Optional[str] = None,
```
Comment (eyurtsev) [inline langchain/document_loaders/telegram.py:85]: @0xRaduan thanks for the contribution. Could you explain why the change of chat_url to chat_entity?

**Comment by eyurtsev** [comment]
@0xRaduan thanks for adding tests!

One of the maintainers (likely me) will move the integration tests into unit tests, so the tests run on CI. 

If you have spare cycles:

1) move integration tests to unit tests
2) mark tests with @pytest.mark.requires("pandas", ...) <-- whatever are the optional packages that are needed
3) `make extended_tests`

File: langchain/document_loaders/telegram.py (hunk 1 above)
Comment (0xRaduan) [reply langchain/document_loaders/telegram.py:85]: Yep, definitely, @eyurtsev.

Per the telethon package, iter_messages(which is used to perform API call to Telegram, to retrieve chat history, [is marked as `Entity`](https://docs.telethon.dev/en/stable/modules/client.html#telethon.client.messages.MessageMethods.iter_messages))

I wanted to add a proper type, but it required me to run poetry lock(so I can get intellisense in the IDE). If you want me, I can do that as well.

Comment (dev2049) [reply langchain/document_loaders/telegram.py:85]: technically a breaking change, but probably ok given this interface is brand new

File: langchain/document_loaders/telegram.py [hunk 2]
```diff
@@ -160,10 +160,10 @@ def find_replies(parent_id: int, reply_data: pd.DataFrame) -> List[int]:
             return all_replies
 
         # Filter out parent messages
-        parent_messages = data[data["is_reply"] is False]
+        parent_messages = data[data["is_reply"] == False]
 
         # Filter out reply messages and drop rows with NaN in 'reply_to_id'
-        reply_messages = data[data["is_reply"] is True].dropna(subset=["reply_to_id"])
+        reply_messages = data[data["is_reply"] == True].dropna(subset=["reply_to_id"])
```
Comment (dev2049) [inline langchain/document_loaders/telegram.py:167]: if is_reply is a boolean column could do just `data[data["is_reply"]]`

File: langchain/document_loaders/telegram.py [hunk 3]
```diff
@@ -217,10 +217,11 @@ def _combine_message_texts(
 
     def load(self) -> List[Document]:
         """Load documents."""
-        if self.chat_url is not None:
+        import pandas as pd
```
Comment (dev2049) [inline langchain/document_loaders/telegram.py:221]: we should keep the import within the try except

Comment (dev2049) [reply langchain/document_loaders/telegram.py:221]: well really we should probably add a separate try/except for when we first use pandas (looks like L239)

Comment (0xRaduan) [reply langchain/document_loaders/telegram.py:221]: I can add it a separate one though, just haven't seen such type of imports in a while.

Comment (0xRaduan) [reply langchain/document_loaders/telegram.py]: fixed

File: langchain/document_loaders/telegram.py [hunk 4]
```diff
@@ -217,24 +218,24 @@ def _combine_message_texts(
...
                 import nest_asyncio
-                import pandas as pd
 
                 nest_asyncio.apply()
                 asyncio.run(self.fetch_data_from_telegram())
             except ImportError:
-                raise ValueError(
-                    "please install with `pip install nest_asyncio`,\
-                                 `pip install nest_asyncio` "
-                )
```
Comment (dev2049) [inline langchain/document_loaders/telegram.py:231]: could me make these ImportErrors and error message "x package not found, please install with..."

Comment (0xRaduan) [reply langchain/document_loaders/telegram.py:231]: Yeah, I agree.

Personally I prefer adding extras, but here after adding telegram extras, it had to trigger update of all deps in the lock file, so I opted-out from doing so.

Comment (0xRaduan) [reply langchain/document_loaders/telegram.py:231]: fixed.

File: tests/unit_tests/document_loaders/test_telegram.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    assert len(docs) == 1
+    assert docs[0].metadata["source"] == str(file_path)
+    assert docs[0].page_content == (
+        "Henry on 2020-01-01T00:00:02: It's 2020...\n\n"
+        "Henry on 2020-01-01T00:00:04: Fireworks!\n\n"
+        "Grace ðŸ§¤ ðŸ\x8d’ on 2020-01-01T00:00:05: You're a minute late!\n\n"
+    )
+
+
+@pytest.mark.requires("telethon", "pandas")
```
Comment (dev2049) [inline tests/unit_tests/document_loaders/test_telegram.py:23]: @eyurtsev look ok?

Comment (eyurtsev) [reply tests/unit_tests/document_loaders/test_telegram.py:23]: Yeah!

**Review by eyurtsev — APPROVED** [review]

**Comment by 0xRaduan** [comment]
Thanks for a rapid review!
